{"name": "Test delete a valid book", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchDriverException: Message: Unable to obtain chromedriver using Selenium Manager; 'str' object has no attribute 'capabilities'; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location", "trace": "service = <selenium.webdriver.chrome.service.Service object at 0x0000026215224740>\noptions = 'C:/Users/annguyenq/Downloads/chromedriver-win64/chromedriver-win64/chromedriver.exe'\n\n    @staticmethod\n    def get_path(service: Service, options: BaseOptions) -> str:\n        path = shutil.which(service.path)\n        try:\n>           path = SeleniumManager().driver_location(options) if path is None else path\n\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\common\\driver_finder.py:42: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.common.selenium_manager.SeleniumManager object at 0x0000026215226750>\noptions = 'C:/Users/annguyenq/Downloads/chromedriver-win64/chromedriver-win64/chromedriver.exe'\n\n    def driver_location(self, options: BaseOptions) -> str:\n        \"\"\"\n        Determines the path of the correct driver.\n        :Args:\n         - browser: which browser to get the driver path for.\n        :Returns: The driver path to use\n        \"\"\"\n    \n        logger.debug(\"Applicable driver not found; attempting to install with Selenium Manager (Beta)\")\n    \n>       browser = options.capabilities[\"browserName\"]\nE       AttributeError: 'str' object has no attribute 'capabilities'\n\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\common\\selenium_manager.py:74: AttributeError\n\nDuring handling of the above exception, another exception occurred:\n\n    @pytest.fixture(scope=\"function\", autouse= True)\n    def setup_driver():\n>       DriverManager.init_driver(pytest.driver_conf)\n\ntestcases\\ui\\conftest.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ncore\\driver\\driver_manager.py:16: in init_driver\n    cls.driver = driver_mapping[config.driver].create_driver(config)\ncore\\driver\\driver_types\\chrome_driver.py:17: in create_driver\n    self._driver = webdriver.Chrome('C:/Users/annguyenq/Downloads/chromedriver-win64/chromedriver-win64/chromedriver.exe')\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\chrome\\webdriver.py:47: in __init__\n    self.service.path = DriverFinder.get_path(self.service, self.options)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nservice = <selenium.webdriver.chrome.service.Service object at 0x0000026215224740>\noptions = 'C:/Users/annguyenq/Downloads/chromedriver-win64/chromedriver-win64/chromedriver.exe'\n\n    @staticmethod\n    def get_path(service: Service, options: BaseOptions) -> str:\n        path = shutil.which(service.path)\n        try:\n            path = SeleniumManager().driver_location(options) if path is None else path\n        except Exception as err:\n>           raise NoSuchDriverException(f\"Unable to obtain {service.path} using Selenium Manager; {err}\")\nE           selenium.common.exceptions.NoSuchDriverException: Message: Unable to obtain chromedriver using Selenium Manager; 'str' object has no attribute 'capabilities'; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location\n\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\common\\driver_finder.py:44: NoSuchDriverException"}, "attachments": [{"name": "stdout", "source": "56aacc2f-07a3-488d-a277-d139aba4bbe9-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "book", "value": "<data_object.book.Book object at 0x0000026215248EF0>"}, {"name": "account", "value": "<data_object.account.Account object at 0x0000026215248EC0>"}], "start": 1702395463625, "stop": 1702395463625, "uuid": "0da6afe2-9059-47e3-9b0c-02acd0bb7bb0", "historyId": "f4ab2b39de66e9e5bd9d4eb662a22187", "testCaseId": "81c9aef85fc2ea69820a2d4d49eabb2c", "fullName": "testcases.ui.test_delete_book#test_delete_book_with_ui", "labels": [{"name": "suite", "value": "Test Delete Book"}, {"name": "parentSuite", "value": "UI Test"}, {"name": "subSuite", "value": "TestDeleteBook"}, {"name": "host", "value": "VNNOT02222"}, {"name": "thread", "value": "14076-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.ui.test_delete_book"}]}